sunkit_image/enhance.py

```diff
@@ -0,0 +1,122 @@
+"""
+This module contains functions that can be used to enhance the entire solar image.
+"""
+
+import numpy as np
+import scipy.ndimage as ndimage
+
+__all__ = [
+    "mgn",
+]
+
+
+def mgn(
+    data,
+    sigma=[1.25, 2.5, 5, 10, 20, 40],
+    k=0.7,
+    gamma=3.2,
+    h=0.7,
+    weights=None,
+    truncate=3
+):
+
+    """
+    Multi-scale Gaussian normalization.
+
+    Parameters
+    ----------
+    data : `numpy.ndarray`
+        Image to be transformed.
+    sigma : `list`, optional
+        Range of Gaussian widths to transform over.
+        Defaults to ``[1.25, 2.5, 5, 10, 20, 40]``.
+    k : `float`, optional
+        Controls the severity of the arctan transformation.
+        Defaults to 0.7
+    gamma : `float`, optional
+        The value used to calculate the  global gamma-transformed image.
+        Ideally should be between 2.5 to 4.
+        Defaults to 3.2
+    h : `float`, optional
+        Weight of global filter to Gaussian filters.
+        Defaults to 0.7
+    weights : `list`, optional
+        Used to weight all the transformed images during the calculation of the
+        final image. If not specificed, all weights are one.
+    width : `int`
+        An odd integer defining the width of the kernel to be convolved.
+    truncate : `int`
+        The number of sigmas to truncate the kernel.
```

Comment:
This needs a little bit more explanation.  Is this full width?

Reply:
I did some research and feel that it is the half width maximum.
![Screenshot from 2019-06-08 23-38-14](https://user-images.githubusercontent.com/39871718/59150765-8f6add80-8a46-11e9-8736-f556a83bd79a.png)
This is place in the paper where they have mentioned it. Maybe you can make it a little more clear.